src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineComplexSchema.java

```diff
@@ -0,0 +1,265 @@
+      } catch (NoAlertPresentException ex) {
+        SeleniumDriver.getDriver().switchTo().newWindow(WindowType.TAB);
+        tryOpenStudioPage();
+      }
+    }
+  }
+
+  private static void tryOpenStudioPage() {
+    SeleniumDriver.openPage(Constants.PIPELINE_STUDIO_URL);
+    WaitHelper.waitForPageToLoad();
+    Helper.setNewSchemaEditor(false);
+  }
+
+  @Then("Add Projection Node")
+  public void addProjectionNode() {
+    Commands.addNodeToCanvas(projectionNode);
+  }
+
+  @Then("Click on Projection node property button")
+  public void clickOnProjectionNodePropertyButton() {
+    CdfStudioActions.navigateToPluginPropertiesPage(projectionNode.getNodeName());
+  }
+
+  @Then("Verify there is one empty row")
+  public void verifyThereIsOneEmptyRow() {
+    WebElement row0Selector = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(schemaRow1));
+
+    Assert.assertTrue(row0Selector.isEnabled());
+  }
+
+  @Then("Type in schema field {string}")
+  public void typeInSchemaField(String field) {
+    WebElement row0Input = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(schemaRow1) + " "
+        + Helper.getCssSelectorByDataTestId(schemaFieldSuffix)
+    );
+    ElementHelper.sendKeys(row0Input, field);
+  }
+
+  @Then("Add a new field row under row one")
+  public void addANewFieldRow() {
+    WebElement addButtonElement = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(schemaRow1) + " "
+        + Helper.getCssSelectorByDataTestId(addButton)
+    );
+    ElementHelper.clickOnElement(addButtonElement);
+  }
+
+  @Then("Verify the added row and type in {string}")
+  public void verifyTheSecondRow(String input) {
+    WebElement schemaRow2Element = Helper.locateElementByTestId(schemaRow2);
+
+    Assert.assertTrue(schemaRow2Element.isDisplayed());
+
+    WebElement schemaRow2FieldElement = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(schemaRow2) + " " + Helper.getCssSelectorByDataTestId(schemaFieldSuffix));
+
+    // Verify the new row is being focused
+    Assert.assertEquals(schemaRow2FieldElement, SeleniumDriver.getDriver().switchTo().activeElement());
+
+    ElementHelper.sendKeys(schemaRow2FieldElement, input);
+  }
+
+  @Then("Verify the previously typed in field")
+  public void verifyThePreviouslyTypedInField() {
+    WebElement anotherFieldElement = Helper.locateElementByTestId("another_field" + schemaFieldSuffix);
+    Assert.assertEquals(anotherFieldElement.getAttribute("value"), "another_field");
+  }
+
+  @Then("Remove row two")
+  public void removeRowTwo() {
+    WebElement removeButtonElement = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(schemaRow2) + " " + Helper.getCssSelectorByDataTestId(removeButton)
+    );
+    ElementHelper.clickOnElement(removeButtonElement);
+  }
+
+  @Then("Verify row three does not exist")
+  public void verifyRowThreeDoesNotExist() {
+    try {
```

Comment:
I think we should use `isElementExists` from helper

Reply:
nice catch, updated